ayon_api/entity_hub.py

```diff
@@ -1485,6 +1486,704 @@ def fill_children_ids(self, children_ids):
+
+        return self._short_name
+
+    def set_short_name(self, short_name):
+        """Change status short name.
+
+        Args:
+            short_name (str): New status short name. 3 letters tops.
+        """
+
+        if not isinstance(short_name, six.string_types):
+            raise TypeError("Short name must be a string.")
+        self._short_name = short_name
+
+    def get_icon(self):
+        """Name of icon to use for status.
+
+        Returns:
+            str: Name of the icon.
+        """
+
+        return self._icon
+
+    def set_icon(self, icon):
+        """Change status icon name.
+
+        Args:
+            icon (str): Name of the icon.
+        """
+
+        if icon is None:
+            icon = ""
+        if not isinstance(icon, six.string_types):
+            raise TypeError("Icon name must be a string.")
+        self._icon = icon
+
+    @property
+    def slugified_name(self):
+        """Slugified and lowere status name.
+
+        Can be used for comparison of existing statuses. e.g. 'In Progress'
+            vs. 'in-progress'.
+
+        Returns:
+            str: Slugified and lower status name.
+        """
+
+        if self._slugified_name is None:
+            self._slugified_name = self.slugify_name(self.name)
+        return self._slugified_name
+
+    def get_state(self):
+        """Get state of project status.
+
+        Return:
+            Literal[not_started, in_progress, done, blocked]: General
+                state of status.
+        """
+
+        return self._state
+
+    def set_state(self, state):
+        """Set color of project status.
+
+        Args:
+            state (Literal[not_started, in_progress, done, blocked]): General
+                state of status.
+        """
+
+        if state not in self.valid_states:
+            raise ValueError("Invalid state '{}'".format(str(state)))
+        self._state = state
+
+    def get_color(self):
+        """Get color of project status.
+
+        Returns:
+            str: Status color.
+        """
+
+        return self._color
+
+    def set_color(self, color):
+        """Set color of project status.
+
+        Args:
+            color (str): Color in hex format. Example: '#ff0000'.
+        """
+
+        if not isinstance(color, six.string_types):
+            raise TypeError(
+                "Color must be string got '{}'".format(type(color)))
+        color = color.lower()
+        if self.color_regex.fullmatch(color) is None:
+            raise ValueError("Invalid color value '{}'".format(color))
+        self._color = color
+
+    name = property(get_name, set_name)
+    short_name = property(get_short_name, set_short_name)
+    project_statuses = property(get_project_statuses, set_project_statuses)
+    index = property(get_index, set_index)
+    state = property(get_state, set_state)
+    color = property(get_color, set_color)
+    icon = property(get_icon, set_icon)
+
+    def _validate_other_p_statuses(self, other):
```

Comment:
Docstrings missing, maybe it could be named `_compare_projects_statuses` ?

Reply:
It's not comparing it's validating...